src/frontend/Typechecker.ml

```diff
@@ -1299,10 +1502,24 @@ let check_tilde_distribution loc tenv id arguments =
       (Promotion.promote_list arguments p, f suffix)
       (* real return type is enforced by [verify_fundef_dist_rt] *)
   | None | Some (SignatureErrors ([], _), _) ->
-      (* Function is non existent *)
-      Semantic_error.invalid_tilde_no_such_dist loc name
-        (List.hd_exn argumenttypes |> snd |> UnsizedType.is_int_type)
-      |> error
+      (* The laplace_marginal helpers are not in the
+         environment lookup, so we patch them back here *)
+      if Stan_math_signatures.is_embedded_laplace_fn (id.name ^ "_lupmf") then
+        let id = {id with name= id.name ^ "_lupmf"} in
+        let fn_expr =
+          check_laplace_fn ~is_cond_dist:true loc cf tenv id arguments in
+        match fn_expr.expr with
+        | CondDistApp (kind, _, args) -> (args, kind)
+        | _ ->
+            Common.ICE.internal_compiler_error
+              [%message
+                "Typechecking a laplace marginal did not return a distribution "
+                  (fn_expr : Ast.typed_expression)]
+      else
+        (* Otherwise, the function is non existent *)
```

Comment:
Can we not add these because they take in tuples of variadic size so they do not fit in the distributions list? It would be nice if we could even make a separate list for them instead of catching a failure